rackhd_BIST/test_utils.py

```diff
@@ -0,0 +1,208 @@
+    try:
+        with open(path) as data_file:
+            lines = data_file.readlines()
+    except IOError:
+        exit_status["message"] = "Can't find or access file {}".format(path)
+        exit_status["exit_code"] = -1
+    else:
+        exit_status["message"] = lines
+    return exit_status
+
+def initiate_logger(name, path):
+    """
+    Initiate bist test logging configures
+    Logs will be stored in log file as well output to console
+    :param name: logger name
+    :param path: log file path
+    :return logger: python logger object with two handlers
+    """
+    logger = logging.getLogger(name)
+    logger.setLevel(logging.DEBUG)
+    time_string = time.strftime("%Y-%m-%d_%H:%M:%S", time.localtime())
+    logfile_name = os.path.join(path, "rackhd_bist_{}.log".format(time_string))
+    logfile = logging.FileHandler(logfile_name)
+    console = logging.StreamHandler()
+    logfile.setLevel(logging.DEBUG)  # Log file will record all messages
+    console.setLevel(logging.WARNING)  # Console will report WARNNING and ERROR
+    formatter = logging.Formatter(
+        '[%(levelname)-7s][%(asctime)-15s] %(message)s'
+    )
+    logfile.setFormatter(formatter)
+    logger.addHandler(logfile)
+    logger.addHandler(console)
+    return logger
+
+def tool_version_compare(version_a, version_b):
+    """
+    Compare for tool version:
+    :param version_a: version to be compared
+    :param version_b: version to be compared
+    :return  1:  version_a is larger than version_b
+            0:  version_a equals version b
+            -1: version-a is smaller than version b
+    """
+    version_a_bits = version_a.split(".")
+    version_b_bits = version_b.split(".")
+    if version_a_bits > version_b_bits:
+        return 1
+    elif version_a_bits < version_b_bits:
+        return -1
+    else:
+        return 0
+
+def get_tool_version(cmd, redirect=False):
+    """
+    Use given command to get tool version
+    :param cmd: command to get tool version
+    :param redirect: flag decide if we should redirect STDERR to STDOUT
+    :return: tool version string
+    """
+    result = robust_check_output(cmd=cmd, shell=False, redirect=redirect)
+    pattern = re.compile(r'\d\.(\d\.)*\d')
+    if result["exit_code"] == 0:
+        version = pattern.search(result["message"])
+        if version:
+            result["message"] = version.group(0) ## refine output message
+        else:
+            result["exit_code"] = -1
+            result["message"] = "Failed to get version with command [{}]".format(",".join(cmd))
+    return result
+
+CONFIGURATION = get_configurations()
+
+class Logger(object):
+    """
+    RackHD BIST specified logging class
+    """
+    def __init__(self):
+        self.logger = initiate_logger(
+            "rackhd_bist_log",
+            CONFIGURATION.get('logPath')
+        )
+
+    def record_log_message(self, description, details, level):
```

Comment:
If `details` is optional but `level` is required, then swap their argument order.